node-graph/gcore/src/vector/generator_nodes.rs

```diff
@@ -32,18 +34,34 @@ fn ellipse_generator(_input: (), radius_x: f64, radius_y: f64) -> VectorData {
 }
 
 #[derive(Debug, Clone, Copy)]
-pub struct RectangleGenerator<SizeX, SizeY> {
+pub struct RectangleGenerator<SizeX, SizeY, IsIndividual, CornerRadius, Clamped> {
 	size_x: SizeX,
 	size_y: SizeY,
+	is_individual: IsIndividual,
+	corner_radius: CornerRadius,
+	clamped: Clamped,
 }
 
 #[node_macro::node_fn(RectangleGenerator)]
-fn square_generator(_input: (), size_x: f64, size_y: f64) -> VectorData {
+fn square_generator<T: Any>(_input: (), size_x: f64, size_y: f64, is_individual: bool, corner_radius: T, clamped: bool) -> VectorData {
```

Comment:
Could I recommend something like:
```rs
trait CornerRadius {
	fn generate(self, size: DVec2, clamped: bool) -> super::VectorData;
}
impl CornerRadius for f64 {
	fn generate(self, size: DVec2, clamped: bool) -> super::VectorData {
		let clamped_radius = if clamped { self.clamp(0., size.x.min(size.y).max(0.) / 2.) } else { self };
		super::VectorData::from_subpaths(vec![Subpath::new_rounded_rect(size / -2., size / 2., [clamped_radius; 4])])
	}
}
impl CornerRadius for [f64; 4] {
	fn generate(self, size: DVec2, clamped: bool) -> super::VectorData {
		let clamped_radius = if clamped {
			self.map(|x| x.clamp(0., size.x.min(size.y).max(0.) / 2.))
		} else {
			self
		};
		super::VectorData::from_subpaths(vec![Subpath::new_rounded_rect(size / -2., size / 2., clamped_radius)])
	}
}
```
Avoiding reliance on the Any trait. Then in the implementation you could just call `corner_radius.generate(DVec2::new(size_x, size_y), clamped)`.

Reply:
Thanks, that seems much cleaner